src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Store.cs

```diff
@@ -128,6 +128,9 @@ public void Add(X509Certificate2 certificate)
         {
             if (certificate == null)
                 throw new ArgumentNullException(nameof(certificate));
+            
+            if (certificate.Handle == IntPtr.Zero)
+                throw new ArgumentNullException(nameof(certificate));
```

Comment:
Are you sure that that is similar code? I see you're just creating a new instance of X509Certificate2, but you're not disposing it. Might be semantically equivalent though, not sure.